### PR TITLE
ICU-22549 Add DateTimePatternGenerator fuzzer

### PR DESCRIPTION
**Pull request by FrankYFTang**
##### Checklist

- [X] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22549
- [X] Required: The PR title must be prefixed with a JIRA Issue number. 
- [X] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [X] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [X] Issue accepted (done by Technical Committee after discussion)
- [X] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: icu4c/source/test/fuzzer/date_time_pattern_generator_fuzzer.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  gen->getSkeleton(fuzzstr, status);
+
+  status = U_ZERO_ERROR;
+  gen->getBaseSkeleton(fuzzstr, status);
+
+  status = U_ZERO_ERROR;
+  gen->getBaseSkeleton(fuzzstr, status);
+
+  status = U_ZERO_ERROR;
+  gen->getPatternForSkeleton(fuzzstr);
```
Comment (richgillam) [inline icu4c/source/test/fuzzer/date_time_pattern_generator_fuzzer.cpp:45]: The thing you _really_ want to be calling (the primary API function for `DateTimePatternGenerator`) is `getBestPattern()`.  I think it's good to call all these other APIs too, but that's the one where the real meat is.

And as in some of these other fuzz tests, if you're just passing a random string here, most of the time you're going to get an error or nonsensical output.  Given the purpose of fuzz testing, this is fine, but there might also be value in additional tests that know more about the expected syntax and play around with it in interesting ways.

**Review by richgillam — APPROVED** [review]
I'll accept that.  Thank you for the addition.

**Comment by richgillam** [comment]
What changed since the last time I approved this?

**Comment by FrankYFTang** [comment]
just resolve merge conflict

**Review by richgillam — APPROVED** [review]
>> What changed since the last time I approved this?
>
>just resolve merge conflict

Okay.  Approved again.

**Comment by FrankYFTang** [comment]
Sorry, my commit message is bad. I need to resquash it again

**Review by richgillam — APPROVED** [review]
Still looks good.  :-)

**Review by echeran — APPROVED** [review]

**Pull request merged**